### ISSUE TITLE
<iterator>

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#5575]
## Issue Details

For several packages we get the warning below   in the testsuite for example [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-96/Linear_cell_complex_Examples/TestReport_lrineau_ArchLinux-CXX17-Release.gz).
```
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22,
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23,
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14,
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42,
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /mnt/testsuite/include/CGAL/Object.h:32,
                 from /mnt/testsuite/include/CGAL/kernel_basic.h:25,
                 from /mnt/testsuite/include/CGAL/basic.h:30,
                 from /mnt/testsuite/include/CGAL/Cartesian/Cartesian_base.h:20,
                 from /mnt/testsuite/include/CGAL/Simple_cartesian.h:20,
                 from /mnt/testsuite/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:20,
                 from /mnt/testsuite/include/CGAL/Linear_cell_complex_traits.h:16,
                 from /mnt/testsuite/include/CGAL/Linear_cell_complex_for_combinatorial_map.h:17,
                 from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Examples/draw_linear_cell_complex.cpp:1:
/usr/include/boost/detail/iterator.hpp:13:1: note: ‘#pragma message: This header is deprecated. Use <iterator> instead.’
   13 | BOOST_HEADER_DEPRECATED("<iterator>")
      | ^~~~~~~~~~~~~~~~~~~~~~~
```
As `<CGAL/basic.h>`  is included by almost all compilations, I would expect this error more often. So my guess is that in the other cases `<iterator>`  gets probably included before.  

I would give it a try and include `<iterator>`  from within `<CGAL/Object.h>`.
Remark: It is not a warning but  a `#pragma` message, but it would be nice to get rid of it. 

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits):  ArchLinux-CXX17-Release
* Compiler:
* Release or debug mode:
* Specific flags used (if any):
* CGAL version:
* Boost version:
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by maxGimeno** [comment]
I found this : https://github.com/boostorg/graph/pull/218

Apparently, Boost deprecated `iterator.hpp` in 1.74 but they forgot to remove their internal inclusion of that header.

**Comment by afabri** [comment]
@maxGimeno you told me that it is fixed in 1.76 ?   If that is the case we should close this issue.

**Comment by maxGimeno** [comment]
I agree
